src/crawlee/storage_clients/_file_system/_request_queue_client.py

```diff
@@ -254,6 +271,7 @@ async def open(
                     metadata=metadata,
                     path_to_rq=path_to_rq,
                     lock=asyncio.Lock(),
+                    recoverable_state=await cls._create_recoverable_state(id=metadata.id, configuration=configuration),
```

Comment:
Instead of creating it three times, we can create it once, store it in a variable, and just pass it where needed.